internal/strings/strings.go

```diff
@@ -99,3 +100,9 @@ func InSlice(a string, list []string) bool {
 	}
 	return false
 }
+
+// WrapUnsafe wraps the provided buffer as a string. The buffer
+// must not be mutated after calling this function.
+func WrapUnsafe(buf []byte) string {
+	return *(*string)(unsafe.Pointer(&buf))
```

Comment:
This is the same code as `strings.Builder`. It provides a safe way to create a string without copying but unfortunately has reasonable overhead of its own due to a safety check on each write. While for most codebases sticking with the safe code is probably for the best, Coraza spends so much time in string processing that I think using `unsafe` in certain parts is probably worth doing